File: src/main/java/at/phe/def/mapreduce/partitioner/HashPartitioner.java
package at.phe.def.mapreduce.partitioner;

/**
 * Author: Philip Heimböck
 * Date: 28.04.16.
 */
public class HashPartitioner implements IPartitioner {

    @Override
    public int partition(Object key, int nrPartitions) {
        int hash = Math.abs(key.hashCode());

        return hash % nrPartitions;
    }
}

File: src/main/dotnet/MapReduce/MapReduceBase/HashPartitioner.cs
using System;

namespace MapReduceBase
{
	public class HashPartitioner : IPartitioner
	{
		#region IPartitioner implementation
		public int Partition (object key, int nrPartitions)
		{
			return Math.Abs (key.GetHashCode()) % nrPartitions;
		}
		#endregion
		
	}
}

